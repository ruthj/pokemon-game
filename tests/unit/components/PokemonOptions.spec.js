import {shallowMount} from '@vue/test-utils'
import PokemonOption from '../../../src/components/PokemonOption'
import {pokemons} from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {

    let wrapper
    
    beforeEach(() => {
        wrapper = shallowMount(PokemonOption, {
            props: {
                pokemons
            }
        })
    })

    test('should match snapshot', () => {
        
        expect (wrapper.html()).toMatchSnapshot()
       
    })

    test('should show four options', () => {
        
        const attributeLi = wrapper.findAll('li')
        expect(attributeLi.length).toBe(4)
        expect(attributeLi[0].text()).toBe('pidgey')
        expect(attributeLi[1].text()).toBe('raticate')
        expect(attributeLi[2].text()).toBe('metapod')
        expect(attributeLi[3].text()).toBe('venusaur')

    })

    test('should $emit selection with paramameters', () => {
        const [li1,li2,li3,li4] = wrapper.findAll('li')
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')
        //console.log(wrapper.emitted('selection'))
        expect(wrapper.emitted('selection').length).toBe(4)
        expect(wrapper.emitted('selection')[0]).toEqual([15])
        expect(wrapper.emitted('selection')[1]).toEqual([19])
        expect(wrapper.emitted('selection')[2]).toEqual([10])
        expect(wrapper.emitted('selection')[3]).toEqual([2])
    })
})