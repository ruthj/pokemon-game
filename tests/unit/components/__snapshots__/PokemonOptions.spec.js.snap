// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component should match snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>pidgey</li>
    <li>raticate</li>
    <li>metapod</li>
    <li>venusaur</li>
  </ul>
</div>
`;
